mergify_cli/ci/cli.py

```diff
@@ -43,23 +46,36 @@
     "--test-language",
     help="Test language",
 )
+@click.option(
+    "--quarantine-check",
+    "--qc",
+    help="""If set to "true", the error code returned by the cli will be `1` if any of the failing tests in the junit file are not quarantined.
+If all the failing tests in the junit file are either a success or quanratined, then the exit code will be `0`.
+""",
+    default="false",
+    show_default=True,
+    type=click.Choice(["true", "false"], case_sensitive=True),
+    envvar="MERGIFY_QUARANTINE_CHECK",
+)
 @click.argument(
     "files",
     nargs=-1,
     required=True,
     type=click.Path(exists=True, dir_okay=False),
 )
 @utils.run_with_asyncio
-async def junit_upload(  # noqa: PLR0913, PLR0917
+async def junit_upload(  # noqa: PLR0913
+    *,
     api_url: str,
     token: str,
     repository: str,
     test_framework: str | None,
     test_language: str | None,
+    quarantine_check: str,
     files: tuple[str, ...],
 ) -> None:
     try:
-        await upload.upload(
+        spans = await upload.upload(
```

Comment:
We should separate the upload from the spans preparation.

Reply:
https://github.com/Mergifyio/mergify-cli/pull/714